client/core/core.go

```diff
@@ -608,6 +608,11 @@ func (dc *dexConnection) reconcileTrades(srvOrderStatuses []*msgjson.OrderStatus
 		previousStatus := trade.metaData.Status
 		newStatus := order.OrderStatus(srvOrderStatus.Status)
 		trade.metaData.Status = newStatus
+		// If there is an associated cancel order, and we are revising the
+		// status of the targeted order, we can infer the cancel order is done.
+		// Update the status of the cancel order and unlink it from the trade.
+		trade.deleteCancelOrder()
```

Comment:
What if we're revising the status to `OrderStatusCanceled`? Wouldn't we be unlinking a correctly linked order?

This would be the case if we crashed after preimage reveal, but before match notifications.